src/commands/watch/buildifier.rs

```diff
@@ -0,0 +1,49 @@
+use std::path::PathBuf;
+use uuid::Uuid;
+
+use super::uppificator::Pause;
+
+pub(crate) struct Buildifier {
+    pub spin_bin: PathBuf,
+    pub manifest: PathBuf,
+    pub clear_screen: bool,
+    pub watched_changes: tokio::sync::watch::Receiver<Uuid>, // TODO: refine which component(s) a change affects
+    pub uppificator_pauser: tokio::sync::mpsc::Sender<Pause>,
+}
+
+impl Buildifier {
+    pub(crate) async fn run(&mut self) {
+        // Other components may close channels as part of shutdown, so if any channels
+        // fail, just exit the loop and fall out normally.
+
+        loop {
+            if self.clear_screen {
+                _ = clearscreen::clear();
+            }
+
+            if self.uppificator_pauser.send(Pause::Pause).await.is_err() {
+                break;
+            }
+
+            _ = self.build_once().await;
```

Comment:
Good point.  Although I'm slightly wary of cancelling builds after some bad experiences locking myself out of a Rust build because of interrupting it as a vital point.  But my wariness is not a good reason for running a lengthy build to completion just to run it all over again.  Time for some more select-related goodness I guess!

Reply:
Hmm, so I tried simulating a Rust program taking a long time to build by having `build.rs` that just slept for 5 seconds.  (I know.  I had to _simulate_ long Rust builds.  You _failed_ me, Rust.)  And cancelling the build just... didn't work.  Neither kill or interrupt stopped it.

I'm hoping this was a weird artifact of using the `build.rs` kludge, because I _know_ I've cancelled Rust builds before.  Or maybe `spin build` was getting killed but not propagating the cancellation to `cargo build`?

Reply:
I think `spin build` might not be propagating the cancellation.  It looks like there's a shell between `spin build` and the actual build process which won't help.

(This is the processes running after I made a change, saved it, made another change, and saved that - notice only one `spin build` but two `cargo build`.  There was also a `cargo check` but I believe that was rust-analyzer catching up with my edits.)

```
ivan@hecate:~$ ps -A -f | egrep "(spin)|(cargo)"
ivan      9724 26324  0 16:32 pts/17   00:00:00 spin watch
ivan     10021  9724  0 16:33 pts/17   00:00:00 /home/ivan/github/spin/target/debug/spin up -f spin.toml
ivan     10034 10021  5 16:33 pts/17   00:00:04 /home/ivan/github/spin/target/debug/spin trigger http
ivan     10543 26323  0 16:34 pts/17   00:00:00 sh -c cargo build --target wasm32-wasi --release
ivan     10544 10543  0 16:34 pts/17   00:00:00 /home/ivan/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/bin/cargo build --target wasm32-wasi --release
ivan     10606  9724  0 16:34 pts/17   00:00:00 /home/ivan/github/spin/target/debug/spin build -f spin.toml
ivan     10616 10606  0 16:34 pts/17   00:00:00 sh -c cargo build --target wasm32-wasi --release
ivan     10617 10616  2 16:34 pts/17   00:00:00 /home/ivan/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/bin/cargo build --target wasm32-wasi --release
```

Reply:
The existing `spin watch` handles this correctly so I guess `watchexec` is killing things more effectively than I am!  I'll have a look and see if I can see what it does.

Reply:
Looks like some magic deep in the bowels of `command_group` which is hopefully reusable.

Reply:
Okay I believe this is now working